src/Injector.ts

```diff
@@ -1,11 +1,22 @@
-import Promise from 'bluebird';
 import { EventEmitter2 } from 'eventemitter2';
-import { EagerSingletonAnnotation } from './annotations';
+import { SingletonAnnotation } from './annotations';
 import { Binder, Module } from './binder';
 import { Key } from './Key';
 import { Provider } from './provider';
 import { Resolver } from './resolver';
 
+const isNil = (p: unknown): p is Exclude<unknown, null | undefined> => {
```

Comment:
This type guard seems incorrect. When I tested this locally, `p` is still shown as `unknown` if it passed the check.
```suggestion
const isNil = (p: unknown): p is null | undefined => {
```